python/lsst/pipe/tasks/hips.py

```diff
@@ -604,3 +614,907 @@ def build_quantum_graph(
+            _initial_ra, _initial_dec = hp.pix2ang(2**max_order, medpix, lonlat=True, nest=True)
+            _initial_fov = hp.nside2resol(2**max_order, arcmin=True)/60.
+
+            if initial_ra is None or initial_dec is None:
+                initial_ra = _initial_ra
+                initial_dec = _initial_dec
+            if initial_fov is None:
+                initial_fov = _initial_fov
+
+        self._write_hips_property_file(
+            hips_base_path,
+            self.config.properties,
+            band,
+            multiband,
+            exposure,
+            max_order,
+            shift_order,
+            area,
+            initial_ra,
+            initial_dec,
+            initial_fov,
+        )
+
+        # Write the MOC coverage
+        self._write_hips_moc_file(
+            hips_base_path,
+            max_order,
+            pixels,
+        )
+
+    def _write_hips_property_file(
+            self,
+            hips_base_path,
+            property_config,
+            band,
+            multiband,
+            exposure,
+            max_order,
+            shift_order,
+            area,
+            initial_ra,
+            initial_dec,
+            initial_fov
+    ):
+        """Write HiPS properties file.
+
+        Parameters
+        ----------
+        hips_base_path : `lsst.resources.ResourcePath`
+            ResourcePath at top of HiPS tree. File will be written
+            to this path as ``properties``.
+        property_config : `lsst.pipe.tasks.hips.HipsPropertyConfig`
+            Configuration for property values.
+        band : `str`
+            Name of band(s) for HiPS tree.
+        multiband : `bool`
+            Is multiband / color?
+        exposure : `lsst.afw.image.Exposure`
+            Sample HPX exposure used for generating HiPS tiles.
+        max_order : `int`
+            Maximum HEALPix order.
+        shift_order : `int`
+            HPX shift order.
+        area : `float`
+            Coverage area in square degrees.
+        initial_ra : `float`
+            Initial HiPS RA position (degrees).
+        initial_dec : `float`
+            Initial HiPS Dec position (degrees).
+        initial_fov : `float`
+            Initial HiPS display size (degrees).
+        """
+        # WARNING: In general PipelineTasks are not allowed to do any outputs
+        # outside of the butler.  This task has been given (temporary)
+        # Special Dispensation because of the nature of HiPS outputs until
+        # a more controlled solution can be found.
+        def _write_property(fh, name, value):
+            """Write a property name/value to a file handle.
+
+            Parameters
+            ----------
+            fh : file handle (blah)
+                Open for writing.
+            name : `str`
+                Name of property
+            value : `str`
+                Value of property
+            """
+            for char in name:
```

Comment:
Checking for unicode spaces?

Reply:
spaces, tabs, anything space-like is NOT ALLOWED by the HiPS standard.

Reply:
Can you say:

```python
import re

if re.search(r"\s", name):
    raise ValueError()
```
rather than looping around? I'd definitely do it as a regex.